unit_tests/integration/test_config.py

```
@@ -372,3 +373,71 @@ def test_20_user_data_format_version_azure(monkeypatch):
+    monkeypatch.setenv("SCT_INSTANCE_TYPE_DB", "i4i.large")
+    monkeypatch.setenv("SCT_CONFIG_FILES", "unit_tests/test_configs/minimal_test_case.yaml")
+    monkeypatch.setenv("SCT_XCLOUD_ENV", "fake-env")
+
+    with (
+        patch("sdcm.sct_config.ScyllaCloudAPIClient", return_value=_make_xcloud_api_mock()),
+        patch("sdcm.sct_config.KeyStore"),
+        patch("sdcm.sct_config.convert_name_to_ami_if_needed", side_effect=lambda v, _: v),
+        patch("sdcm.sct_config.find_scylla_repo", return_value="https://fake-repo/scylla.repo"),
+    ):
+        conf = sct_config.SCTConfiguration()
+        with pytest.raises(ValueError, match="xcloud_replication_factor .* cannot be greater than n_db_nodes"):
```

Ruff (0.15.17)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)